### PR TITLE
ScanCode: Tolerate underscores in license keys

### PR DESCRIPTION
**Pull request by fviernau**

### CODE REVIEW

File: scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt [hunk 1]
```diff
@@ -128,6 +128,8 @@ private fun getLicenseFindings(result: JsonNode): List<LicenseFinding> {
     files.flatMapTo(licenseFindings) { file ->
         val licenses = file["licenses"]?.asSequence().orEmpty()
         licenses.map {
+            println("XXXXXXXXXXXXXXXXXXXXXxXXX")
+            println("license: " + it)
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt:132]: Left-over debug code.

File: scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt [hunk 2]
```diff
@@ -128,6 +128,8 @@ private fun getLicenseFindings(result: JsonNode): List<LicenseFinding> {
     files.flatMapTo(licenseFindings) { file ->
         val licenses = file["licenses"]?.asSequence().orEmpty()
         licenses.map {
+            println("XXXXXXXXXXXXXXXXXXXXXxXXX")
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt:131]: Commit message nit: The "I'll investigate" part does not really fit to the passive wording we use in commit messages and could be reworded.

File: scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt [hunk 3]
```diff
@@ -262,7 +264,7 @@ private fun generateScannerOptions(options: JsonNode?): String {
 }
 
 /**
- * Map messages about unknown issues to a more compact form. Return true if solely memory errors occurred,
+ * Map messages about unknown isorg.ossreviewtoolkit.scanner.scanners.ScanCodeResultParserKt.getLicenseFindingssues to a more compact form. Return true if solely memory errors occurred,
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt:267]: Unintended change.

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt (hunk 2 above)
Comment (fviernau) [reply scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt:131]: Done.

File: scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt (hunk 1 above)
Comment (fviernau) [reply scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt:132]: Done.

File: scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt (hunk 3 above)
Comment (fviernau) [reply scanner/src/main/kotlin/scanners/ScanCodeResultParser.kt:267]: Done.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**